pkg/ccl/changefeedccl/metrics.go

```diff
@@ -62,6 +64,20 @@ type AggMetrics struct {
 	}
 }
 
+type metricsRecorderBuilder func(requiresCostAccounting bool) metricsRecorder
+
+type metricsRecorder interface {
```

Comment:
nice. A demonstration of an adage that "all problem can be solved by indirection".